internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,510 @@
+}
+
+func (h *releaseSourcesHandler) insertToDB(tx *gorm.DB, dbReleaseImages []*common.ReleaseImage) error {
+	h.log.Debugf("Inserting %d records to openshift_versions table", len(dbReleaseImages))
+
+	if len(dbReleaseImages) == 0 {
+		return nil
+	}
+
+	return tx.Create(dbReleaseImages).Error
+}
+
+// removeReleaseImageDuplicates removes release images such that only one release image
+// of each version, CPU architecture will remain
+func (h *releaseSourcesHandler) removeReleaseImageDuplicates(releases []*common.ReleaseImage) ([]*common.ReleaseImage, error) {
+	releaseImagesMap := map[string]*common.ReleaseImage{}
+
+	for _, releaseImage := range releases {
+		duplicateDBReleaseImage, exists := releaseImagesMap[*releaseImage.URL]
+		if exists {
+			// If the release image that already exists or its diplicate are labeled as defualt,
+			// label the one already considered as default
+			if duplicateDBReleaseImage.Default || releaseImage.Default {
+				duplicateDBReleaseImage.Default = true
+			}
+
+			// If the release image that aleady exists has support level beta (as it came from candidate channel),
+			// replace its support level with the other release's support level
+			if duplicateDBReleaseImage.SupportLevel == models.OpenshiftVersionSupportLevelBeta {
+				duplicateDBReleaseImage.SupportLevel = releaseImage.SupportLevel
+			}
+
+			continue
+		}
+
+		releaseImagesMap[*releaseImage.URL] = releaseImage
+	}
+
+	mappingResult := funk.Map(releaseImagesMap, func(url string, dbReleaseImage *common.ReleaseImage) *common.ReleaseImage {
+		return dbReleaseImage
+	})
+	if dbReleaseImages, ok := mappingResult.([]*common.ReleaseImage); ok {
+		return dbReleaseImages, nil
+	}
+
+	return nil, errors.New("error occurred while trying to convert the map to slice during removal of release image duplicates")
+
+}
+
+func (h *releaseSourcesHandler) getDynamicReleaseImages() ([]*common.ReleaseImage, error) {
+	// OCP releases API needs amd64, arm64 instead of x86_64, aarch64 respectively
+	cpuArchMapToAPIArch := map[string]string{
+		common.X86CPUArchitecture:     common.AMD64CPUArchitecture,
+		common.AARCH64CPUArchitecture: common.ARM64CPUArchitecture,
+	}
+
+	// switch back
+	cpuArchMapFromAPIArch := map[string]string{
+		common.AMD64CPUArchitecture: common.X86CPUArchitecture,
+		common.ARM64CPUArchitecture: common.AARCH64CPUArchitecture,
+	}
+
+	releases := []*common.ReleaseImage{}
+	// Aggregate slice of releases from releases API
+	for _, releaseSource := range h.releaseSources {
+		openshiftVersion := *releaseSource.OpenshiftVersion
+		for _, upgradeChannel := range releaseSource.UpgradeChannels {
+			cpuArchitecture := *upgradeChannel.CPUArchitecture
+			for _, channel := range upgradeChannel.Channels {
+				apiCpuArchitecture, shouldSwitch := cpuArchMapToAPIArch[cpuArchitecture]
+				if shouldSwitch {
+					cpuArchitecture = apiCpuArchitecture
+				}
+
+				graph, err := h.releasesClient.getReleases(channel, openshiftVersion, cpuArchitecture)
```

Comment:
As far as i understand this is your client no? In this case i think better to make 
apiCpuArchitecture, shouldSwitch := cpuArchMapToAPIArch[cpuArchitecture] inside of it.
In anycase i don't understand why we need to change it back as here 
`if arch, ok := cpuArchMapFromAPIArch[cpuArchitecture]; ok {
						cpuArchitecture = arch
					}`
why not to simply use apiCpuArchitecture as param to h.releasesClient.getReleases(channel, openshiftVersion, cpuArchitecture ?